spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/ExpressionEvaluatingSqlParameterSourceFactory.java

```diff
@@ -148,6 +168,12 @@ public void afterPropertiesSet() throws Exception {
 		this.getEvaluationContext().setVariable("staticParameters", this.staticParameters);
 	}
 
+	private void enrichSqlParametersType(AbstractSqlParameterSource sqlParameterSource) {
+		if (this.sqlParametersTypes != null) {
+			this.sqlParametersTypes.forEach((parameter, type) -> sqlParameterSource.registerSqlType(parameter, type));
```

Comment:
Seems for me this can be method reference - `sqlParameterSource::registerSqlType`

Reply:
I did thought the same, but I felt the current one more readable. Feel free to change during merge :)

Reply:
Bdw out of curiosity, what type of warning it causes?

Reply:
![image](https://user-images.githubusercontent.com/1071590/32449369-5e0469ee-c2df-11e7-91f7-1373f81b2203.png)

That's enough to obtain my attention and distract from other stuff around.